packages/frontend/src/components/common/Table/BodyCell.tsx

```diff
@@ -23,97 +24,72 @@ interface CommonBodyCellProps {
+            if (!onSelectCell || !hasContextMenu || !hasData) return;
 
-    const hasContextMenu = hasData && !!CellContextMenu;
+            e.stopPropagation();
+            e.preventDefault();
 
-    const handleSelect = () => {
-        if (!hasContextMenu) return;
-        onSelect();
-    };
-
-    const handleDeselect = () => {
-        onDeselect();
-    };
+            const cellWithResultValue = cell as Cell<ResultRow, ResultRow[0]>;
+            onSelectCell(cellWithResultValue, e.currentTarget);
+        },
+        [onSelectCell, hasContextMenu, hasData, cell],
+    );
 
     return (
-        <Popover2
-            isOpen={selected}
-            lazy
-            minimal
-            position={Position.BOTTOM_RIGHT}
-            hasBackdrop
-            backdropProps={{ onClick: handleDeselect }}
-            onOpening={() => handleSelect()}
-            onClose={() => handleDeselect()}
-            content={
-                CellContextMenu && (
-                    <CellContextMenu
-                        cell={cell as Cell<ResultRow, ResultRow[0]>}
-                    />
-                )
-            }
-            renderTarget={({ ref: ref2, ...popoverProps }) => (
-                <Tooltip2
-                    lazy
-                    position={Position.TOP}
-                    disabled={!tooltipContent || minimal}
-                    content={tooltipContent}
-                    renderTarget={({ ref: ref1, ...tooltipProps }) => (
-                        <Td
-                            ref={mergeRefs(ref1, ref2)}
-                            {...(tooltipProps as any)}
-                            {...popoverProps}
-                            className={className}
-                            style={style}
-                            $rowIndex={index}
-                            $isSelected={selected}
-                            $isLargeText={isLargeText}
-                            $isMinimal={minimal}
-                            $isInteractive={hasContextMenu}
-                            $isCopying={copying}
-                            $backgroundColor={backgroundColor}
-                            $fontColor={fontColor}
-                            $hasData={hasContextMenu}
-                            $isNaN={!hasData || !isNumericItem}
-                            onClick={selected ? handleDeselect : handleSelect}
-                            onKeyDown={onKeyDown}
-                        >
-                            <RichBodyCell
-                                cell={cell as Cell<ResultRow, ResultRow[0]>}
-                            >
-                                {children}
-                            </RichBodyCell>
-                        </Td>
-                    )}
-                />
-            )}
-        />
+        <Tooltip
+            withinPortal
+            position="top"
+            disabled={!tooltipContent || minimal}
+            label={tooltipContent}
+        >
```

Comment:
could we also do something about the rendering of these tooltips? 
For example, similar to the `Menu` position calculation or maybe having one `Tooltip` shared across the cells? 🤔